app/services/oauth_clients/connection_manager.rb

```diff
@@ -91,7 +90,8 @@ def refresh_token
     # @param scope (OAuth2 RFC) specifies the resources to access. Nextcloud has only one global scope.
     def get_authorization_uri(scope: [], state: nil)
       client = rack_oauth_client # Configure and start the rack-oauth2 client
-      client.authorization_uri(scope:, state:)
+      joined_scopes = @config.compute_scopes(scope)
```

Comment:
🟡 I tend to question the need of merging scopes. Do we have the use case, that we use the same `oauth_configuration` on two different places with two different scopes? It would restrict the interface of that method and make it simpler to use.

Reply:
You have a point. I preserved the ability to pass in more scopes as this could come from the outside of the `ConnectionManager`, but that is not a present use case.

I'll remove this and make the `CM` ask the config about the scopes.